libpod/container_internal_linux.go

```diff
@@ -1383,6 +1383,11 @@ func (c *Container) generateResolvConf() (string, error) {
 		return "", err
 	}
 
+	// Determine if symlink points to any of the systemd-resolved files
+	if strings.HasPrefix(resolvPath, "/run/systemd/resolve/") {
+		resolvPath = "/run/systemd/resolve/resolv.conf"
+	}
```

Comment:
Just a nit-pick: The logic for retrieving the `resolvPath` seems to be scattered around in the implementation and I would prefer having it in a own function.